go/types/opcache.go

```diff
@@ -58,27 +105,33 @@ func (p *opCache) Set(mapKey Value, mapVal Value) {
 		// +-----------------------+---------------------+----------------------+
 
 		// Note that, if mapKey and/or mapVal are prolly trees, any in-memory child chunks will be written to vrw at this time.
-		p.ldbKeyScratch[0] = byte(mapKey.Type().Kind())
-		copy(p.ldbKeyScratch[1:], mapKey.Hash().DigestSlice())
-		mapKeyData := encToSlice(mapKey, p.keyScratch[:], p.vrw)
-		mapValData := encToSlice(mapVal, p.valScratch[:], p.vrw)
-
-		mapKeyByteLen := len(mapKeyData)
-		data := make([]byte, uint32Size+mapKeyByteLen+len(mapValData))
-		binary.LittleEndian.PutUint32(data, uint32(mapKeyByteLen))
-		copy(data[uint32Size:], mapKeyData)
-		copy(data[uint32Size+mapKeyByteLen:], mapValData)
+		ldbKeyArray := [uint32Size + 1 + hash.ByteLen]byte{}
+		binary.LittleEndian.PutUint32(ldbKeyArray[:], opc.colId)
+		ldbKeyArray[uint32Size] = byte(mapKey.Type().Kind())
+		ldbKey := ldbKeyArray[0 : uint32Size+1]
+		ldbKey = append(ldbKey, mapKey.Hash().DigestSlice()...)
```

Comment:
Using `copy()` here reads much more clearly, to me. Possibly busting this out into a helper method like `constructLdbKey(scratch []byte, colId uint32, mapKey Value)` would make the code here in MapSet more readable.